multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ApplicationEnvironmentCalculator.java

```diff
@@ -0,0 +1,94 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.common.util.MapUtil;
+import org.cloudfoundry.multiapps.controller.api.model.ProcessType;
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
+import org.cloudfoundry.multiapps.controller.core.model.DeployedMtaApplication;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+import com.sap.cloudfoundry.client.facade.CloudOperationException;
+
+@Named
+public class ApplicationEnvironmentCalculator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApplicationEnvironmentCalculator.class);
+
+    private final DeploymentTypeDeterminer deploymentTypeDeterminer;
+
+    @Inject
+    public ApplicationEnvironmentCalculator(DeploymentTypeDeterminer deploymentTypeDeterminer) {
+        this.deploymentTypeDeterminer = deploymentTypeDeterminer;
+    }
+
+    public Map<String, String> calculateNewApplicationEnv(ProcessContext context, CloudApplicationExtended newApp) {
+        if (!(newApp.getAttributesUpdateStrategy()
+                    .shouldKeepExistingEnv()
+            && deploymentTypeDeterminer.determineDeploymentType(context)
+                                       .equals(ProcessType.BLUE_GREEN_DEPLOY))) {
+            return newApp.getEnv();
+        }
+        if (context.getVariable(Variables.DEPLOYED_MTA) == null) {
+            context.getStepLogger()
+                   .debug(Messages.INITIAL_DEPLOYMENT_WILL_NOT_SEARCH_FOR_LIVE_APPLICATION);
+            return newApp.getEnv();
+        }
```

Comment:
Since these return the same result and I do not think the log message is that important, can we inline these conditions so it can be cleaner?

```suggestion
        if (isInitialDeploy() || !isBlueGreenDeploy(context) 
            || !newApp.getAttributesUpdateStrategy()
                      .shouldKeepExistingEnv()            
            return newApp.getEnv();
        }
        //...
        }
        
        private boolean isInitialDeploy() {
            return context.getVariable(Variables.DEPLOYED_MTA) == null;
        }
        
        private boolean isBlueGreenDeploy(ProcessContext context) {
            return ProcessType.BLUE_GREEN_DEPLOY.equals(deploymentTypeDeterminer.determineDeploymentType(context));
        }
```